tests/example_test.go

```diff
@@ -381,6 +397,16 @@ func TestExample(t *testing.T) {
 				_, _, err = runCommand(t, orgDir, "git", "clone", "--depth", "1", parsed.cloneURL)
 				require.NoError(t, err, "cloning repo")
 			}
+
+			// If we have a specific commit, use that.
+			if tt.commit != "" {
+				_, _, err := runCommand(t, repoDir, "git", "fetch", "--depth", "1", "origin", tt.commit)
+				require.NoError(t, err, "fetching commit: %s", tt.commit)
+				_, _, err = runCommand(t, repoDir, "git", "checkout", tt.commit)
+				require.NoError(t, err, "checking out commit: %s", tt.commit)
+				_, _, err = runCommand(t, repoDir, "git", "reset", "--hard")
+				require.NoError(t, err, "git reset")
```

Comment:
This `git reset --hard` will not do anything.  Since we start from a clean copy, `git checkout` will do the right thing here and checkout the right files.